wordCreator.py

```diff
@@ -0,0 +1,294 @@
+                indices.append(i)
+
+                #Add the value to un-normed value to the running sum
+                lastVal += tempList[i]
+
+        #Return the list of ranges and the original indices of the ranges
+        return ranges, indices
+
+
+    def getWeightedLetter(self, probList, inSum):
+        """
+        Receives list of probabilities (probList) and sum of original entities
+        of the list
+        Generates random letter based on how the letter is weighted in the list
+        returns that number
+        """
+        print(inSum)
+        #Get probability ranges and their indices
+        ranges, indices = self.getRanges(probList, inSum)
+        randLetterIndex = random.randint(1, inSum)
+        lastVal = 0
+        #Loop through values in ranges
+        for i in range(len(ranges)):
+            #If random integer is between the previous and current range
+            if lastVal < randLetterIndex and randLetterIndex <= ranges[i]:
+                #return the character at that index
+                return chr(indices[i])
+            #Reinitialize last value
+            lastVal = ranges[i]
+
+
+    def genWord(self, n):
+        """
+        Generates random word based on transition list of length n
+        Returns a word (word)
+        """
+        #Initiealize method variables
+        word = []
+        wordLen = n
+
+        #Create normalized list of sums and get the total characters
+        normList, totalChar = self.norm(self.sumList)
+
+        #Get a letter based on the frequency of letters in the set
+        word.append(self.getWeightedLetter(normList, totalChar))
+
+        #loop until you hit the end of the word length
+        for i in range(wordLen-1):
+            #append random character based on the previous character
+            letterIndex = ord(word[i])
+            flag = True
+            subtractor = 1
+            while flag:
+                if self.sumList[letterIndex]:
+                    word.append(self.getWeightedLetter(self.transitions[letterIndex], self.sumList[letterIndex]))
+                    flag = False
+                else:
+                    letterIndex  = ord(word[i-subtractor])
+                    subtractor+=1
+
+
+
+        #return word as a string
+        return ''.join(word)
+
+    def genRandWord(self):
+        """
+        Generates random word based on transition list
+        Returns a word (word)
+        Same as genWord(), but with a random word length
+        """
+        wordLen = random.randint(self.minLetters, self.maxLetters)
+        word = []
+        normList, totalChar = self.norm(self.sumList)
+        word.append(self.getWeightedLetter(normList, totalChar))
+        for i in range(wordLen-1):
+            letterIndex = ord(word[i])
+            word.append(self.getWeightedLetter(self.transitions[letterIndex], self.sumList[letterIndex]))
+        return ''.join(word)
+
+    def __main__(self):
```

Comment:
According to python naming convention, double underscore ( __ ) is reserved for specific [cases](http://stackoverflow.com/questions/1301346/what-is-the-meaning-of-a-single-and-a-double-underscore-before-an-object-name).